packages/wallet/backend/src/paymentPointer/service.ts

```diff
@@ -144,20 +151,38 @@ export class PaymentPointerService implements IPaymentPointerService {
           throw new NotFound('Web monetization asset not found.')
         }
 
-        assetScale = webMonetizationAsset.scale
-        assetCode = webMonetizationAsset.code
-      }
+        const rafikiPaymentPointer =
+          await this.deps.rafikiClient.createRafikiPaymentPointer(
+            args.publicName,
+            webMonetizationAsset.id,
+            url
+          )
 
-      paymentPointer = await PaymentPointer.query().insert({
-        url: rafikiPaymentPointer.url,
-        publicName: args.publicName,
-        accountId: args.accountId,
-        id: rafikiPaymentPointer.id,
-        assetCode,
-        assetScale,
-        balance: 0,
-        isWM: args.isWM
-      })
+        paymentPointer = await PaymentPointer.query().insert({
+          url: rafikiPaymentPointer.url,
+          publicName: args.publicName,
+          accountId: args.accountId,
+          id: rafikiPaymentPointer.id,
+          isWM: args.isWM,
+          assetCode: webMonetizationAsset.code,
+          assetScale: webMonetizationAsset.scale
+        })
+      } else {
+        const rafikiPaymentPointer =
+          await this.deps.rafikiClient.createRafikiPaymentPointer(
+            args.publicName,
+            account.assetId,
+            url
+          )
+
+        paymentPointer = await PaymentPointer.query().insert({
+          url: rafikiPaymentPointer.url,
+          publicName: args.publicName,
+          accountId: args.accountId,
+          id: rafikiPaymentPointer.id,
+          isWM: args.isWM
+        })
+      }
```

Comment:
Any suggestions on how we can make this more readable? I had to split it like this because the WM payment pointer was getting created with the original asset code and scale (USD/2) instead of USD/9.

Reply:
```
      let webMonetizationAsset;
      if (args.isWM) {
        //* Web monetization feature requires an asset with scale MAX_ASSET_SCALE to exist. It's default assetCode is USD for now
        webMonetizationAsset =
            await this.deps.rafikiClient.getRafikiAsset(
                'USD',
                this.deps.env.MAX_ASSET_SCALE
            )

        if (!webMonetizationAsset) {
          throw new NotFound('Web monetization asset not found.')
        }
      }

      const assetId = webMonetizationAsset?.id || account.assetId
        const rafikiPaymentPointer =
          await this.deps.rafikiClient.createRafikiPaymentPointer(
            args.publicName,
              assetId,
            url
          )

        paymentPointer = await PaymentPointer.query().insert({
          url: rafikiPaymentPointer.url,
          publicName: args.publicName,
          accountId: args.accountId,
          id: rafikiPaymentPointer.id,
          isWM: args.isWM,
          assetCode: webMonetizationAsset?.code,
          assetScale: webMonetizationAsset?.scale
        })

      args.isWM &&
        (await this.deps.cache.set(paymentPointer.id, paymentPointer, {
          expiry: 60
        }))
```